### PR TITLE
Fix example of kubectl create configmap command

### PR DESCRIPTION
**Pull request by geraldmayr**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:
Fixes an example shown at https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#-em-configmap-em-

#### Does this PR introduce a user-facing change?

NONE

### CODE REVIEW

**Review by draveness — COMMENTED** [review]
/ok-to-test
/lgtm

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_configmap.go [hunk 1]
```diff
@@ -66,7 +66,7 @@ var (
 		  kubectl create configmap my-config --from-literal=key1=config1 --from-literal=key2=config2
 
 		  # Create a new config map named my-config from the key=value pairs in the file
-		  kubectl create configmap my-config --from-file=path/to/bar
+		  kubectl create configmap my-config --from-env-file=path/to/bar
```
Comment (rikatz) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_configmap.go:69]: hum the old comment seems right to me.

--from-env-file is covered in the example below

Comment (geraldmayr) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_configmap.go:69]: Hi @rikatz,

File "bar"
```
    a=a
    b=b
    c=c
```

When we run `kubectl create configmap my-config --from-file=bar --dry-run=client -o yaml` we get the following yaml:
```
apiVersion: v1
data:
  bar: |
    a=a
    b=b
    c=c
kind: ConfigMap
metadata:
  creationTimestamp: null
  name: my-config
```

So it does not interpret the key=value pairs within the file as mentioned in the old comment, while `kubectl create configmap my-config --from-env-file=bar --dry-run=client -o yaml` does:
```
apiVersion: v1
data:
  a: a
  b: b
  c: c
kind: ConfigMap
metadata:
  creationTimestamp: null
  name: my-config
```

In my opinion, the old comment is wrong or least misleading.

Comment (KnVerey) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_configmap.go:69]: I agree that it is misleading, and I'm not sure why we need both examples. Both were introduced in the original `--from-env-file` PR. In that PR, reviewers [specifically asked](https://github.com/kubernetes/kubernetes/pull/38882#discussion_r101799633) for `.env` to be used in the examples. And the equivalent for `secret` only has the one example both originally and today. I suggest removing the additional misleading example.

**Pull request closed without merging**